os/include/tinyara/binfmt/binfmt.h

```diff
@@ -100,6 +104,19 @@ struct binary_s {
 #ifdef CONFIG_APP_BINARY_SEPARATION
 	struct mm_heap_s *uheap;	/* User heap pointer to allocate memory for sections */
 	uint32_t uheap_size;		/* The size of user heap */
+	size_t textsize;		/* Size of text section */
+	size_t rosize;			/* Size of ro section */
+	size_t datasize;		/* Size of data section */
+	uint32_t ramstart;		/* Start address of ram partition */
+	uint32_t ramsize;		/* Size of the RAM paritition */
+	uint32_t heapstart;		/* Start address of app heap area */
+#endif
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD
+	uint32_t reload;		/* Indicate whether this binary will be reloaded */
```

Comment:
How about checking the load/reload status with `data_backup`? If `data_backup` is not NULL, it means reload case.

Reply:
I am currently checking binp for reload scenario. If binp is not NULL, then it is reload scenario. data_backup is a variable inside binp. So, I think checking for binp is ok.